### PR TITLE
Feature/user api

### PR DESCRIPTION
**Pull request by tanu456**

### CODE REVIEW

**Review by kiranbhanushali — APPROVED** [review]

File: models/user.model.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ const UserSchema = new mongoose.Schema(
     phone_number: { type: String, required: true },
     donations: { type: Array, default: [] },
     aadhar_number: { type: String, required: true },
+    isVerified: { type: Boolean, default: false },
+    emailtoken: { type: String },
```
Comment (akshitbansal108) [inline models/user.model.js:14]: since we are using _ separated words, is_verified and email_token are more apt.

**Comment by akshitbansal108** [comment]
Since we have used "Donation" as schema name, do update the file names in controller to donation.controller.js

File: controllers/user.controller.js [hunk 2]
```diff
@@ -13,3 +22,172 @@ exports.getAllUsers = async (req, res, next) => {
     });
   }
 };
+
+// @desc    Get User by query params
+// @route   GET /api/v1/users/get?field=value&....
+exports.getUsersByQuery = async (req, res, next) => {
+  var users = null;
```
Comment (akshitbansal108) [inline controllers/user.controller.js:29]: why do we need to define var users as null, we can directly use const user in try block

Comment (kiranbhanushali) [reply controllers/user.controller.js:29]: i guess both will work same

Comment (akshitbansal108) [reply controllers/user.controller.js:29]: Yes they will.

File: models/user.model.js (hunk 1 above)
Comment (tanu456) [reply models/user.model.js:14]: okay I will make the changes and update PR

**Review by PragsJ — APPROVED** [review]

**Pull request merged**